src/object.hpp

```diff
@@ -498,6 +524,229 @@ class object_view final : public readable_object<object_view> {
 
 static_assert(sizeof(object_view) == sizeof(void *));
 
+class array_view {
+public:
+    array_view() = default;
+
+    // NOLINTNEXTLINE(google-explicit-constructor, hicpp-explicit-conversions)
+    array_view(const detail::object *o)
+    {
+        if (o == nullptr || o->type != object_type::array) {
+            throw std::invalid_argument("array_view initialised with null or incompatible type");
+        }
+        data_ = o->via.array.ptr;
+        size_ = o->via.array.size;
+    }
+    // NOLINTNEXTLINE(google-explicit-constructor, hicpp-explicit-conversions)
+    array_view(const detail::object &o) : array_view(&o) {}
+    // NOLINTNEXTLINE(google-explicit-constructor, hicpp-explicit-conversions)
+    array_view(object_view o) : array_view(o.ptr()) {}
+    // NOLINTNEXTLINE(google-explicit-constructor, hicpp-explicit-conversions)
+    array_view(const owned_object &ow);
+    // NOLINTNEXTLINE(google-explicit-constructor, hicpp-explicit-conversions)
+    array_view(const borrowed_object &ow);
```

Comment:
Repetitive... maybe using some concept for types convertible to detail::object *? If not `std::convertible_to` because you don't want to make it directly convertible, some other method to expose the pointer.

Reply:
I'll address this in the next PR.